src/Kiota.Builder/PathSegmenters/CommonPathSegmenter.cs

```diff
@@ -19,13 +19,14 @@ protected CommonPathSegmenter(string rootPath, string clientNamespaceName)
         protected static string GetLastFileNameSegment(CodeElement currentElement) => currentElement.Name.Split('.').Last();
         public string GetPath(CodeNamespace currentNamespace, CodeElement currentElement) {
             var fileName = NormalizeFileName(currentElement);
-            var namespacePathSegments = currentNamespace.Name
+            var namespacePathSegments = new List<string>(currentNamespace.Name
                                             .Replace(ClientNamespaceName, string.Empty)
                                             .TrimStart('.')
-                                            .Split('.')
-                                            .Union(GetAdditionalSegment(currentElement, fileName))
-                                            .Where(x => !string.IsNullOrEmpty(x))
-                                            .Select(x => NormalizeNamespaceSegment(x));
+                                            .Split('.'));
+            namespacePathSegments.AddRange(GetAdditionalSegment(currentElement, fileName)); //Union removes duplicates
```

Comment:
The comment suggests that you wish to remove duplicates but the call to `Union()` has been removed. Yes?

Reply:
thanks, updated the comment 75b6a74 (push pending)